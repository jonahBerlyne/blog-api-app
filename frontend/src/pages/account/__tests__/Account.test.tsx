import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter as Router } from "react-router-dom";
import * as ReactRouter from 'react-router';
import AccountActivationPage from "../[slug]";
import axios from "axios";
import { act } from "react-dom/test-utils";

jest.mock('axios');

describe("Account Activation Page", () => {
 const useParams = jest.spyOn(ReactRouter, 'useParams');

 beforeEach(() => {
  useParams.mockReturnValue({
   slug: 'slugExample'
  });
 });

 afterEach(() => {
  jest.resetAllMocks();
 });
 
 const setup = async () => {  
  const { container } = render(
   <Router>
    <AccountActivationPage />
   </Router>
  );

  const resolve = Promise.resolve();
  await act(async () => {
   await resolve;
  });

  return {
   container
  };
 }

 it("renders the account activation page", async () => {
  (axios.post as jest.MockedFunction<typeof axios.post>).mockResolvedValue({
   data: {
    msg: 'Success Msg Example'
   }
  });

  const { container } = await setup();

  expect(container).toMatchSnapshot();
 });

 it("shows an account activation success msg", async () => {
  (axios.post as jest.MockedFunction<typeof axios.post>).mockResolvedValue({
   data: {
    msg: 'Success Msg Example'
   }
  });

  await setup();

  expect(screen.queryByTestId('errMsg')).not.toBeInTheDocument();
  expect(screen.getByTestId('successMsg').innerHTML).toEqual('Success Msg Example');
 });

 it("shows an account activation error msg", async () => {
  (axios.post as jest.MockedFunction<typeof axios.post>).mockRejectedValue({
   response: {
    data: {
     msg: new Error('Error Msg Example').message
    }
   }
  });

  await setup();

  expect(screen.queryByTestId('successMsg')).not.toBeInTheDocument();
  expect(screen.getByTestId('errMsg').innerHTML).toEqual('Error Msg Example');
 });
});