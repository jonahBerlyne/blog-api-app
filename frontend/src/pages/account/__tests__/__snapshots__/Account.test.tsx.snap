// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Activation Page renders the account activation page 1`] = `
<div>
  <div>
    <div
      class="successMsg"
      data-testid="successMsg"
    >
      Success Msg Example
    </div>
  </div>
</div>
`;
